integration_test.go

```diff
@@ -0,0 +1,85 @@
+package main_test
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	main "github.com/lawrencegripper/tfoperatorbridge"
+)
+
+var _ = Describe("When working with a resource group", func() {
+	name := "tftest-" + strings.ToLower(main.RandomString(12))
+	gvrResourceGroup := schema.GroupVersionResource{
+		Group:    "azurerm.tfb.local",
+		Version:  "valpha1",
+		Resource: "resource-groups",
+	}
+
+	It("should allow the resource lifecycle", func() {
+		By("creating the resource-group CRD")
+
+		Expect(k8sClient).ToNot(BeNil())
+
+		obj := unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "azurerm.tfb.local/valpha1",
+				"kind":       "resource-group",
+				"metadata": map[string]interface{}{
+					"name": name,
+				},
+				"spec": map[string]interface{}{
+					"name":     name,
+					"location": "westeurope",
+				},
+			},
+		}
+		options := metav1.CreateOptions{}
+		_, err := k8sClient.Resource(gvrResourceGroup).Namespace("default").Create(context.TODO(), &obj, options)
+		Expect(err).To(BeNil())
+
+		By("returning the resource ID")
```

Comment:
Oh nice, I'd not used the `by` func before

Reply:
Yeah, I'm still finding my feet a bit with ginkgo, but `By` let me keep some output showing the flow but control the ordering :-)